e3/job/__init__.py

```diff
@@ -81,14 +81,23 @@ def __lt__(self, other):
         """Compare two jobs.
 
         This method can be reimplemented in order to provider other
-        priority systems.
+        priority systems. Default implementation assume all jobs have
+        the same priority.
 
         :param other: another Job
         :type other: Job
         :return: True if self has a strictly higher priority than other
         :rtype: bool
         """
-        return self.index < other.index
+        return False
+
+    def __eq__(self, other):
+        """Check for equality.
+
+        Implementation is needed in order to be able to compare tuples
+        containing Job objects.
```

Comment:
That's too strange ;)

As discussed live, introduce a priority attribute and use it in e3.job.scheduler.